src/logger/__tests__/MetricsContext.test.ts

```diff
@@ -353,6 +390,20 @@ test.each([
   }).not.toThrow(InvalidMetricError);
 });
 
+test('put metric with same key and different resolution in single flush throws error' ,() => {
+  //arrange
+  const context = MetricsContext.empty();
+  const expectedKey = 'MetricName';
+  const expectedValue = faker.datatype.number();
+  const expectedUnit = 'None';
+
+  // act 
+ expect(() => {
+    context.putMetric(expectedKey, expectedValue,expectedUnit,StorageResolution.High);
```

Comment:
`context.putMetric(expectedKey, expectedValue,expectedUnit,StorageResolution.High)` should not be expected to throw an error.

Reply:
Updated!